deploy/cephcsi/image/Dockerfile

```diff
@@ -0,0 +1,17 @@
+
+FROM centos:7
```

Comment:
Why not using `ceph/ceph` image instead? See: https://hub.docker.com/r/ceph/ceph/tags

Reply:
I would be happy to make that change if other ceph-csi experts agree. @rootfs @ShyamsundarR @gman0 ?

Reply:
@lseb I like the general idea, but an unaware if the ceph/ceph image is unduly bigger than what we require in this case.

This image needs "ceph-common kmod ceph-fuse attr e2fsprogs xfsprogs rbd-nbd" as in the yum install below.

The ceph/ceph image seems to contain iSCSI, NFS Ganesha etc, which are not needed in this case.

Also, the image size is ~228MB as compared to the current ceph-csi images running at ~150MB (just pointing it out, not attempting to save bits and bytes at these sizes yet).

The ceph image is better from an operational standpoint, as ensuring it has the right dependencies is done by the image itself, hence more apt to use the same.

The above is what I notice, @leseb your thoughts on these and if they matter much?

Reply:
@ShyamsundarR the image size, as well as the unnecessary packages, are a valid point. However, it doesn't hurt much, I'm just trying to avoid using home-made images that could use wrong sources. The images we build on `ceph/ceph` are Ceph's official image, supported and maintained by the Ceph team.

I just realized that our current image does not contain `rbd-nbd` and `ceph-fuse`, this can be easily fixed but I don't want to hold to PR for that. We can still fix this later.

Btw why do we need `rbd-nbd`, AFAIK we don't use NBD but KRBD, can someone clarify? Thanks!

Reply:
> @ShyamsundarR the image size, as well as the unnecessary packages, are a valid point. However, it doesn't hurt much, I'm just trying to avoid using home-made images that could use wrong sources. The images we build on `ceph/ceph` are Ceph's official image, supported and maintained by the Ceph team.

I agree, the extra size does not really bother me, also staying on the official image is the right option.

> 
> I just realized that our current image does not contain `rbd-nbd` and `ceph-fuse`, this can be easily fixed but I don't want to hold to PR for that. We can still fix this later.
> 
> Btw why do we need `rbd-nbd`, AFAIK we don't use NBD but KRBD, can someone clarify? Thanks!

NBD can be used via a [`mounter`](https://github.com/ceph/ceph-csi/blob/da52e8cc63348fed9595c61807bd1bcb83020b13/examples/rbd/storageclass.yaml#L52) option provided, and so we support it. I am not sure about why we do and if we should carry it forward, but possibly a different discussion. As you have started the process of including it in the Ceph official image, it should be good for now.

Reply:
https://github.com/ceph/ceph-container/pull/1337 has just recently merged, which should allow us to use ceph/ceph as a base.

Reply:
Not yet, we haven't built the image yet.